tests/posix/common/src/mqueue.c

```diff
@@ -97,4 +96,201 @@ ZTEST(mqueue, test_mqueue)
+{
+	mqd_t mqd;
+	struct mq_attr attrs = {
+		.mq_msgsize = MESSAGE_SIZE,
+		.mq_maxmsg = MESG_COUNT_PERMQ,
+	};
+	struct sigevent not = {
+		.sigev_notify = SIGEV_NONE,
+		.sigev_value.sival_ptr = (void *)&notification_executed,
+		.sigev_notify_function = notify_function_basic,
+	};
+	int32_t mode = 0777;
+	int flags = O_RDWR | O_CREAT;
+
+	notification_executed = false;
+	memset(rec_data, 0, MESSAGE_SIZE);
+
+	mqd = mq_open(queue, flags, mode, &attrs);
+
+	zassert_false(mq_notify(mqd, &not),
+		      "Unable to set notification.");
+
+	zassert_false(mq_send(mqd, send_data, MESSAGE_SIZE, 0),
+		      "Not able to send message");
+
+	zassert_true(notification_executed,
+		     "Notification not triggered.");
+
+	zassert_false(mq_close(mqd),
+		      "Unable to close message queue descriptor.");
+	zassert_false(mq_unlink(queue), "Not able to unlink Queue");
+}
+
+void notify_function_thread(union sigval val)
+{
+	mqd_t mqd;
+	pthread_t sender = (pthread_t)val.sival_int;
+
+	zassert_not_equal(sender, pthread_self(),
+			  "Notification function should be executed from different thread.");
+
+	mqd = mq_open(queue, O_RDONLY);
+
+	mq_receive(mqd, rec_data, MESSAGE_SIZE, 0);
+	zassert_false(strcmp(rec_data, send_data),
+		      "Error in data reception. exp: %s act: %s", send_data, rec_data);
+
+	zassert_false(mq_close(mqd),
+		      "Unable to close message queue descriptor.");
+
+	notification_executed = true;
+}
+
+ZTEST(mqueue, test_mqueue_notify_thread)
+{
+	mqd_t mqd;
+	struct mq_attr attrs = {
+		.mq_msgsize = MESSAGE_SIZE,
+		.mq_maxmsg = MESG_COUNT_PERMQ,
+	};
+	struct sigevent not = {
+		.sigev_notify = SIGEV_THREAD,
+		.sigev_value.sival_int = (int)pthread_self(),
+		.sigev_notify_function = notify_function_thread,
+	};
+	int32_t mode = 0777;
+	int flags = O_RDWR | O_CREAT;
+
+	notification_executed = false;
+	memset(rec_data, 0, MESSAGE_SIZE);
+
+	mqd = mq_open(queue, flags, mode, &attrs);
+
+	zassert_false(mq_notify(mqd, &not),
+		      "Unable to set notification.");
+
+	zassert_false(mq_send(mqd, send_data, MESSAGE_SIZE, 0),
+		      "Not able to send message");
+
+	usleep(USEC_PER_MSEC * 10U);
+
+	zassert_true(notification_executed,
+		     "Notification not triggered.");
+
+	zassert_false(mq_close(mqd),
+		      "Unable to close message queue descriptor.");
+	zassert_false(mq_unlink(queue), "Not able to unlink Queue");
+}
+
+ZTEST(mqueue, test_mqueue_notify_non_empty_queue)
+{
+	mqd_t mqd;
+	struct mq_attr attrs = {
+		.mq_msgsize = MESSAGE_SIZE,
+		.mq_maxmsg = MESG_COUNT_PERMQ,
+	};
+	struct sigevent not = {
+		.sigev_notify = SIGEV_NONE,
+		.sigev_value.sival_ptr = (void *)&notification_executed,
+		.sigev_notify_function = notify_function_basic,
+	};
+	int32_t mode = 0777;
+	int flags = O_RDWR | O_CREAT;
+
+	notification_executed = false;
+	memset(rec_data, 0, MESSAGE_SIZE);
+
+	mqd = mq_open(queue, flags, mode, &attrs);
+
+	zassert_false(mq_send(mqd, send_data, MESSAGE_SIZE, 0),
+		      "Not able to send message");
```

Comment:
This and others, to be consistent with other lines that used "Unable"
```suggestion
		      "Unable to send message");
```